internal/venice-common/src/main/proto/VeniceReadService.proto

```diff
@@ -29,4 +54,126 @@ message VeniceServerResponse {
 
   uint32 errorCode = 6;
   string errorMessage = 7;
+}
+
+message SingleGetRequest {
+  string resourceName = 1;
+  uint32 partition = 2;
+  string key = 3;
+  bool isRetryRequest = 4;
+}
+
+message SingleGetResponse {
+  int32 statusCode = 1;
+  bytes value = 2;
+  sint32 schemaId = 3;
+  uint32 compressionStrategy = 4;
+  optional string errorMessage = 5;
+  string contentType = 6;
+  uint32 contentLength = 7;
+  uint32 rcu = 8;
+}
+
+message MultiGetRequest {
+  string resourceName = 1;
+  uint32 partition = 2;
+  bytes  keyBytes = 3;            // used for batch get
+  uint32 keyCount = 4;
+  bool isRetryRequest = 5;
+}
+
+message MultiGetResponse {
+  int32 statusCode = 1;
+  bytes value = 2;
```

Comment:
> so that it can be written out incrementally, while all (or most?) 


I don't follow. Can you please elaborate?

Reply:
The idea was that the header-type content should be a prefix, and the repetitive content can be flushed to the wire incrementally. And therefore, the ordering might be better if the values are at the end... but since these new protocols are still a work in progress, it's fine to ignore for now...